packages/webpack-cli/lib/plugins/InteractiveModePlugin.js

```diff
@@ -0,0 +1,242 @@
+const readline = require('readline');
+const { red, green, cyanBright, bold } = require('colorette');
+const logger = require('../utils/logger');
+const { SyncHook } = require('tapable');
+let version;
+try {
+    version = require('webpack').version;
+} catch (err) {
+    logger.error(err);
+    process.exit(2);
+}
+
+/**
+ * Displays command space at bottom of screen
+ * @param {string} msg message to print with command
+ * @param {boolean} status currently watching or not
+ */
+const spawnCommand = (msg, status) => {
+    const lines = 3;
+    const totalRows = process.stdout.rows;
+    readline.cursorTo(process.stdout, 0, totalRows - lines);
+    readline.clearScreenDown(process.stdout);
+
+    console.log(bold(cyanBright(`ⓘ  ${msg}`)));
+    process.stdout.write('\n');
+
+    readline.cursorTo(process.stdout, 0, totalRows - 2);
+
+    // for current status
+    if (status) {
+        process.stdout.write(`${green('⬤')}  `);
+    } else {
+        process.stdout.write(`${red('⬤')}  `);
+    }
+};
+
+/**
+ * Clear the whole terminal
+ */
+const clrscr = () => {
+    process.stdout.write('\x1B[2J\x1B[3J\x1B[H');
+};
+
+/**
+ * Helper plugin for child compilers if MultiCompiler is supplied
+ */
+class InteractiveModeMultiCompilerHelperPlugin {
+    constructor() {
+        this.name = 'InteractiveModeMultiCompilerHelperPlugin';
+    }
+
+    apply(compiler) {
+        // clear terminal if any one of child starts compilation
+        compiler.hooks.beforeCompile.tap(this.name, () => {
```

Comment:
Let's add test with `--profile` here, anyway I think `beforeCompile` is good hook, I think we need improve logic for multi compiler mode like:
```js
this.isCleanScreen = false;

// Logic

for (const compiler of compilers) {
  compiler.hooks.beforeCompile.tap(this.name, () => {
    if (this.isCleanScreen) { 
      clrscr();
    }
  });
}  
```

to avoid racing between two/three/etc compilations

Reply:
Other idea to avoid `isCleanScreen`, `allDone` and etc, to keep status, i.e. `this.status = 'before-compile' | 'compile' | 'after-done' | 'pause' | 'resume'`; (maybe be more, maybe you will know better here), so the plugin will know current status and will not do extra things, also using the one status variables will be comfortably to handle multi compiler mode